test/addr_book_suite.go

```diff
@@ -331,3 +360,88 @@ func testPeersWithAddrs(m pstore.AddrBook) func(t *testing.T) {
 		})
 	}
 }
+
+func testCertifiedAddresses(m pstore.AddrBook) func(*testing.T) {
+	return func(t *testing.T) {
+		cab := m.(pstore.CertifiedAddrBook)
+
+		priv, _, err := test.RandTestKeyPair(crypto.Ed25519, 256)
+		if err != nil {
+			t.Errorf("error generating testing keys: %v", err)
+		}
+
+		id, _ := peer.IDFromPrivateKey(priv)
+		allAddrs := GenerateAddrs(10)
+		certifiedAddrs := allAddrs[:5]
+		uncertifiedAddrs := allAddrs[5:]
+		rec := peer.NewPeerRecord()
+		rec.PeerID = id
+		rec.Addrs = certifiedAddrs
+		signedRec, err := rec.Sign(priv)
```

Comment:
That's a lot of assignments! Enclosing these in a `var ()` block could help readability, due to alignment.